sdk/core/Azure.Core.TestFramework/src/TestProxy.cs

```diff
@@ -204,8 +214,18 @@ private static bool TryParsePort(string output, string scheme, out int? port)
             return false;
         }
 
-        public void CheckForErrors()
+        public async Task CheckProxyOutputAsync()
         {
+            // add a small delay to allow the log output for the just finished test to be collected into the _output StringBuilder
+            await Task.Delay(20);
```

Comment:
This shouldn't be a noticeable difference - for every 3000 tests in a single test assembly, we could end up taking 1 minute longer. It only scales for tests per assembly as multiple assemblies will be run in parallel, in general. If this proves to be a problem, we could restrict this to only when debug logging is enabled.

Reply:
Ended up putting this behind a single flag.